src/txn/txn.c

```diff
@@ -1979,6 +2014,12 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
           "durable timestamp");
     }
 
+    /*
+     * Leave the commit generation after all the updates are processed and the snapshot is released.
+     */
+    if (!prepare)
+        __wt_session_gen_leave(session, WT_GEN_TXN_COMMIT);
```

Comment:
We can leave the generation after __wt_txn_release(session); No need to hold this longer.

Reply:
change the comment to snapshot is released

Reply:
Fixed in 9ef616b393614af333f6c1f0660018ebab2effef